mantidproject/mantid
# New version of reflectometry `Auto` algorithms

New versions of some Reflectometry algorithms:

- `ReflectometryReductionOne` and `ReflectometryReductionOneAuto`.
- `CreateTransmissionWorkspace` and `CreateTransmissionWorkspaceAuto`.
- `SpecularReflectionPositionCorrect`

**To test:**

- Check that new algorithms have their own unit tests (and check that the new test files are testing the new version).
- Check that no functionality was added/removed in the old versions (i.e. check files `ReflectometryReductionOne`, `ReflectometryReductionOneAuto`, `CreateTransmissionWorkspace`, `CreateTransmissionWorkspaceAuto`, `SpecularReflectionPositionCorrect` and ensure code changes are minimal, both headers and source files). Only some tests were added.
- Check that each of the new algorithms comes with an .rst file. Build the documentation and make sure it is accurate.
- System test `RROAutoFunctionalityTests` was removed. The reason is that it uses a decorator, `construct_standard_algorithm()` that made impossible selecting the old version of the algorithm. Check that those tests have been moved to `ReflectometryReductionOneAutoTest`.
- Check that release notes were updated.
- Check that existing system tests are testing version 1 of the algorithms.
- Run this script: [ReflectometryReductionOneAuto_comparison.zip](https://github.com/mantidproject/mantid/files/708726/ReflectometryReductionOneAuto_comparison.zip). It is the script I've been using to compare versions 1 and 2 of `ReflectometryReductionOneAuto` (it also tests indirectly the other algorithms mentioned above). Plot the `IvsLam` workspaces generated by each version. They should be compatible within error bars and should be very similar (some differences are expected in some cases, such as POLREF data, specially if data are noisy).
- Code review: have a look at the new classes listed above (I also added a new version of `ReflectometryWorkflowBase` although it does not have unit tests).
- Check that the old interface is using version 1 of the algorithms (this will be changed in PR https://github.com/mantidproject/mantid/pull/18239)
- Check that the new interface is using version 2 of the algorithms. Test that the interface works reducing a few datasets. I've checked with the .tbl we have for INTER. Check that results look exactly the same as in master. Also, check that the parameters (such as the angle, Q min, Qmax, etc) are updated in the table when the reduction is complete.

Fixes #17429.
Fixes #18029.

Release notes: https://github.com/mantidproject/mantid/pull/18058/commits/b0bea6d9828431f5a53bd472268f3a9154a49a4a and https://github.com/mantidproject/mantid/pull/18058/commits/83ca77c53c6ea58ec76859237431ee9a95d089c7

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
